watershed_workflow/utils.py

```diff
@@ -579,6 +599,73 @@ def create_empty_raster(target_bounds, crs, target_dx, dtype, nodata):
     out = nodata * np.ones((height, width), dtype)
     return out_profile, out
 
+def CrossProduct(A):
+     
+    # Stores coefficient of X
+    # direction of vector A[1]A[0]
+    X1 = (A[1][0] - A[0][0])
+ 
+    # Stores coefficient of Y
+    # direction of vector A[1]A[0]
+    Y1 = (A[1][1] - A[0][1])
+ 
+    # Stores coefficient of X
+    # direction of vector A[2]A[0]
+    X2 = (A[2][0] - A[0][0])
+ 
+    # Stores coefficient of Y
+    # direction of vector A[2]A[0]
+    Y2 = (A[2][1] - A[0][1])
+ 
+    # Return cross product
+    return (X1 * Y2 - Y1 * X2)
+ 
+
+def isConvex(points):
```

Comment:
can this be replaced by:

```
def convex_hull(points):
    poly = shapely.geometry.Polygon(points) # is points array or polygon?
    return poly.convex_hull.area == poly.area
```

shapely's convex_hull should be very fast and will compute the convex polygon containing all points.  This would be better than maintaining our own algorithm

Reply:
Thanks for the suggestion. Using this now.